src/codecs/cats/ldpc.c

```diff
@@ -0,0 +1,91 @@
+#include <assert.h>
+#include <string.h>
+
+#include "ldpc.h"
+#include "ldpc_matrices.h"
+
+#define GET_BIT(byte, bit) (((byte) & (1 << (7-(bit)))) != 0)
+#define SET_BIT(byte, bit) (byte |= 1 << 7-bit)
+#define CLEAR_BIT(byte, bit) (byte &= ~(1 << 7-bit))
+#define FLIP_BIT(byte, bit) (byte ^= (1 << (7 - bit)))
+
+cats_ldpc_code_t *cats_ldpc_pick_code(int len)
+{
+    if(128 <= len)
+        return &tm2048;
+    else if(32 <= len)
+        return &tc512;
+    else if(16 <= len)
+        return &tc256;
+    else
+        return &tc128;
+}
+
+// returns # of bytes written to parity_out
+size_t cats_ldpc_encode_chunk(uint8_t *data, cats_ldpc_code_t *code, uint8_t *parity_out)
+{
+    // Code parameters
+    int k = code->k;
```

Comment:
Not being familiar with the LDPC algorithm, would `k`, `r` and `b` be somehow self-explanatory for one who knows about them? I've tried to write code in this project so that it should be _mostly_ possible to read the code on its own -- so that the code documents itself. Could we possibly use some more descriptive variable names anyway (also in the struct)?

Also links to the algorithm documentation (or CATS documentation) would be great!